### PR TITLE
Email Resource

### PR DESCRIPTION
**Pull request by cromigon**
### Email Resource
Adding the Email Resource to glesys-go

Checklist
- [X] email/overview (with filtering and pagination)
- [x] email/globalquota
- [x] email/list
- [x] email/editaccount
- [x] email/delete
- [x] email/createaccount
- [x] email/quota
- [x] email/createalias
- [x] email/editalias
- [x] email/costs

# Examples

## Overview
Fetching overview without filtering or pagination
```go
// Fetch overview
overview, err := client.Emails.Overview(context.Background(),
    glesys.OverviewParams{})

if err != nil {
    fmt.Println("Error fetching overview: ", err)
}
```

Fetching overview with filtering
```go
// Fetch overview
overview, err := client.Emails.Overview(context.Background(),
    glesys.OverviewParams{
        Filter: "example.com",
})

if err != nil {
    fmt.Println("Error fetching overview: ", err)
}
```

Fetching overview with pagination
```go
// Fetch overview
overview, err := client.Emails.Overview(context.Background(),
    glesys.OverviewParams{
        Page: 2,
})

if err != nil {
    fmt.Println("Error fetching overview: ", err)
}
```

Fetching overview with filtering and pagination
```go
// Fetch overview
overview, err := client.Emails.Overview(context.Background(),
    glesys.OverviewParams{
        Filter: "example.com",
        Page: 2,
})

if err != nil {
    fmt.Println("Error fetching overview: ", err)
}
```

## Global Quota
Fetch global quota:
```go
global, err := client.Emails.GlobalQuota(context.Background(),
    glesys.GlobalQuotaParams{})
```

Set new global quota:
```go
global, err := client.Emails.GlobalQuota(context.Background(),
    glesys.GlobalQuotaParams{
        GlobalQuota: 20480,
})
```

## List
List Emails for domain:
```go
list, err := client.Emails.List(context.Background(), "example.com", 
    glesys.ListEmailsParams{})

if err != nil {
    fmt.Println("Error fetching email list: ", err)
}
```

List Emails with domain and filtering:
```go
list, err := client.Emails.List(context.Background(), "example.com", 
    glesys.ListEmailsParams{
        Filter: "user@example.com"
    })

if err != nil {
    fmt.Println("Error fetching email list: ", err)
}
```

## EditAccount
```go
edit, err := client.Emails.EditAccount(context.Background(), "user@your-domain.com", 
    glesys.EditAccountParams{
        AntiSpamLevel:      3,
	AntiVirus:          "yes",
	Password:           "SuperSecretPassword",
	AutoRespond:        "yes",
	AutoRespondMessage: "Your Automatic Response",
	Quota:              400,
	RejectSpam:         "yes",
    })

if err != nil {
    fmt.Println("Error editing account: ", err)
}
```

## Delete
```go
client.Emails.Delete(context.Background(), "user@your-domain.com")
```

## CreateAccount
```go
create, err := client.Emails.CreateAccount(context.Background(),
    glesys.CreateAccountParams{
        AccountName:                "new_user@your-domain.com",
        AntiSpamLevel:               3,
	AntiVirus:                         "yes",
	Password:                        "SuperSecretPassword",
	AutoRespond:                 "yes",
	AutoRespondMessage: "Your Automatic Response",
	Quota:                               400,
	RejectSpam:                     "yes",
    })

if err != nil {
    fmt.Println("Error creating account: ", err)
}
```

## Email Account Quota
```go
quota, err := client.Emails.Quota(context.Background(), "user@example.com")

if err != nil {
    fmt.Println("Error fetching email account quota: ", err)
}
```

## Create Alias
```go
alias, err := client.Emails.CreateAlias(context.Background(),
  glesys.EmailAliasParams{
    EmailAlias: "alias@your-domain.com,
    GoTo: "user@your-domain.com",
  })

if err != nil {
    fmt.Println("Error creating email alias: ", err)
}
```

## Edit Alias
```go
alias, err := client.Emails.EditAlias(context.Background(),
  glesys.EmailAliasParams{
    EmailAlias: "alias@your-domain.com,
    GoTo: "another_user@your-domain.com",
  })

if err != nil {
    fmt.Println("Error updating email alias: ", err)
}
```

## Get Costs
```go
costs, err := client.Emails.Costs(context.Background())

if err != nil {
    fmt.Println("Error fetching email costs: ", err)
}
```

### CODE REVIEW

File: doc_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func ExampleEmailService_CreateAccount() {
 
 	// NOTE: All parameters except for EmailAccount and Password are optional and can be omitted.
 	createaccount, _ := client.Emails.CreateAccount(context.Background(), glesys.CreateAccountParams{
-		EmailAccount:       "new_user@example.com",
+		EmailAccount:       "new_user@your-domain.com",
```
Comment (brother) [inline doc_test.go]: I rather have `example.com` than `your-domain.com`, not the most important thing in the world though.

File: emails_test.go [hunk 2]
```diff
@@ -209,3 +209,71 @@ func TestCreateAccount(t *testing.T) {
...
+	assert.Equal(t, 3, costs.Costs.Accounts.Amount, "costs.accounts.amount is correct")
+	assert.Equal(t, 0, costs.Costs.Accounts.Cost.Amount, "costs.accounts.cost.amount is correct")
+	assert.Equal(t, "SEK", costs.Costs.Accounts.Cost.Currency, "costs.accounts.cost.currency is correct")
+	assert.Equal(t, "1.00", costs.PriceList.Quota.Amount, "pricelist.quota.amount is correct")
+	assert.Equal(t, "SEK", costs.PriceList.Quota.Currency, "pricelist.quota.currency is correct")
+	assert.Equal(t, "GB", costs.PriceList.Quota.Unit, "pricelist.quota.unit is correct")
+	assert.Equal(t, 10, costs.PriceList.Quota.FreeAmount, "pricelist.quota.freeamount is correct")
+	assert.Equal(t, 50, costs.PriceList.Accounts.Amount, "pricelist.accounts.amount is correct")
+	assert.Equal(t, "SEK", costs.PriceList.Accounts.Currency, "pricelist.accounts.currency is correct")
+	assert.Equal(t, "50 accounts", costs.PriceList.Accounts.Unit, "pricelist.accounts.unit is correct")
```
Comment (brother) [inline emails_test.go]: This is pretty awkward. We should adjust the API output here...

File: doc_test.go (hunk 1 above)
Comment (cromigon) [reply doc_test.go]: That makes sense, I will update it. `example.com` is already used in `ExampleServerService_Edit` so this change will make the domain used within the test consistent.

File: emails_test.go (hunk 2 above)
Comment (brother) [reply emails_test.go]: This correction is now live. The Unit is 'account'.

Comment (cromigon) [reply emails_test.go]: Thank you for the notification! I'll push an update to the tests so that they reflect the change.

File: client.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ type Client struct {
 	project    string
 	userAgent  string
 
+	Emails          *EmailService
```
Comment (wecc) [inline client.go]: I see where this pattern is coming from however I would like to suggest we call this `EmailDomains` instead. This service does not represent _emails_ but rather the _domains with email enabled/activated_ (amongst other things).

Even though there are technical details linking the `email/*` and `domain/*` parts of our API together they are not really the same _thing_ and we've been trying to differentiate them using the terms "DNS Domains" and "Email Domains".

**Comment by wecc** [comment]
Wow, this is awesome. Thank you so much @cromigon!

Other than my suggestion to rename `client.Emails => client.EmailDomains` this is ✅ from me!

File: client.go (hunk 3 above)
Comment (cromigon) [reply client.go]: Thank you for the feedback, I agree that `EmailDomains` is a better name in the larger perspective. I'll push an update soon to change this. 👍

File: doc.go [hunk 4]
```diff
@@ -20,6 +20,7 @@
 // The different modules of the GleSYS API are available on the client.
 // For example:
 //
+//     client.Emails.Overview(...)
```
Comment (wecc) [inline doc.go]:
```suggestion
//     client.EmailDomains.Overview(...)
```

Comment (cromigon) [reply doc.go]: Ah, totally missed that one, thank you for updating it for me 👍

**Pull request merged**